polkadot/node/network/approval-distribution/src/lib.rs

```diff
@@ -488,11 +507,17 @@ struct BlockEntry {
 	knowledge: Knowledge,
 	/// A votes entry for each candidate indexed by [`CandidateIndex`].
 	candidates: Vec<CandidateEntry>,
+	/// Information about candidate metadata.
+	candidates_metadata: Vec<(CandidateHash, CoreIndex, GroupIndex)>,
 	/// The session index of this block.
 	session: SessionIndex,
 	/// Approval entries for whole block. These also contain all approvals in the case of multiple
 	/// candidates being claimed by assignments.
 	approval_entries: HashMap<(ValidatorIndex, CandidateBitfield), ApprovalEntry>,
+	/// The block vrf story.
+	pub vrf_story: RelayVRFStory,
```

Comment:
mixing pub and private fields is not a good sign, do we really need a getter and setter here? seems only a getter is needed, so I'd add it instead of making it pub